### ISSUE TITLE
Fedora support by adding :Z suffix option to docker --volume

### ISSUE THREAD

**Issue by vorburger** [andrewparmet/buf-gradle-plugin#21]
```
❯ ./gradlew bufLint

> Task :app:bufLint
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
Resolving "bufbuild/buf" using unqualified-search registries (/etc/containers/registries.conf)
Trying to pull registry.fedoraproject.org/bufbuild/buf:0.54.1...
Trying to pull registry.access.redhat.com/bufbuild/buf:0.54.1...
Trying to pull docker.io/bufbuild/buf:0.54.1...
Getting image source signatures
Copying blob sha256:229ec212b546059c5669d1addfce96ea3dc25db2e5c6eeb94d73df54a1f06d0c
Copying blob sha256:a07461379343c0fd0dec23271d4583c31e689c5efc372876359f7bef71214a26
Copying blob sha256:5843afab387455b37944e709ee8c78d7520df80f8d01cf7f861aae63beeddb6b
Copying blob sha256:5843afab387455b37944e709ee8c78d7520df80f8d01cf7f861aae63beeddb6b
Copying blob sha256:a07461379343c0fd0dec23271d4583c31e689c5efc372876359f7bef71214a26
Copying blob sha256:229ec212b546059c5669d1addfce96ea3dc25db2e5c6eeb94d73df54a1f06d0c
Copying config sha256:a15cd6451bffd2f7ab5f609c4b66576eb75035879771fc52d1a9111dce9ed060
Writing manifest to image destination
Storing signatures
Failure: stat /workspace/buf.work.yaml: permission denied.

> Task :app:bufLint FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:bufLint'.
> Process 'command 'docker'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
3 actionable tasks: 3 executed
```

**Comment by vorburger** [comment]
The `Failure: stat /workspace/buf.work.yaml: permission denied.` is because I'm using https://podman.io instead of https://www.docker.com on Fedora Linux (with the `podman-docker` package, which aliases `docker` to `podman`, as the `Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.` message reveals.)

Podman supports [SELinux](https://en.wikipedia.org/wiki/Security-Enhanced_Linux) on Fedora, which causes it to be a PITA; repro:

```
$ podman run --volume "$(pwd):/workspace" --workdir /workspace bufbuild/buf lint
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
Failure: could not clean relative path "".
```

As per https://github.com/containers/podman/issues/3683#issuecomment-517239831 (also https://github.com/moby/moby/issues/30934), this can be solved like this:

    $ podman run --volume "$(pwd):/workspace:Z" --workdir /workspace bufbuild/buf lint

Note the `:Z` suffix option to `docker --volume`. Would you consider adding this `:Z`? It should cause no problems on `docker` instead of `podman` on Linux distributions without SELinux, as it should be ignored there.

**Comment by vorburger** [comment]
Until this is fixed, there is a workaround:

    $ podman run --volume "$(pwd):/workspace:z" --workdir /workspace bufbuild/buf lint

Note the lower- instead of upper-case `:z`suffix - this applies an SELinux label allowing access by all containers - and `buf-gradle-plugin` works. This is more of a "hack" than a "solution" though - the Gradle plugin should just work out of the box on Fedora (and other Linux distributions with SELinux enabled), as the proposed PR would allow it to.

**Comment by andrewparmet** [comment]
Thanks for the report and the contribution! Published in 0.2.3.

**Comment by andrewparmet** [comment]
@vorburger If you're feeling adventurous I just published a new version that doesn't use Docker. Feel free to try it out and report back if you have any issues.
